pkg/loader/refs.go

```diff
@@ -244,5 +245,7 @@ func (c *TypeChecker) check(root *Package) {
 	// ...then, we can safely type-check ourself
 	root.NeedTypesInfo()
 
+	c.Lock()
+	defer c.Unlock()
```

Comment:
Do we want to synchronize writes only or reads and writes?
There is a read in the same function.

Reply:
yep, need to synchronize reads as well

Reply:
Fixed it!